src/plugins/plot/spectralAggregatePlot/SpectralAggregatePlotCompositionPolicy.js

```diff
@@ -40,10 +40,14 @@ export default function SpectralAggregatePlotCompositionPolicy(openmct) {
         return metadata.values().length > 0 && hasAggregateDomainAndRange(metadata);
     }
 
+    function hasNoChildren(parentObject) {
+        return parentObject.composition && parentObject.composition.length < 1;
+    }
+
     return {
         allow: function (parent, child) {
             if ((parent.type === SPECTRAL_AGGREGATE_KEY)
-                && ((child.type !== 'telemetry.plot.overlay') && (hasSpectralAggregateTelemetry(child) === false))
+                && ((child.type !== 'telemetry.plot.overlay') && (hasSpectralAggregateTelemetry(child) === false) || (hasNoChildren(parent) === false))
```

Comment:
I thought we didn't want to put this check into the composition policy?

Reply:
Yes, but your code in the view to check and remove the object from composition reminded me that we should prevent adding the object rather than having to remove it later.